worker/crlWorker/crlWorker.go

```diff
@@ -0,0 +1,200 @@
+			Result:     nil,
+		}
+		return
+	}
+
+	// Check if our certificate is in the revoked list
+	revoked := certList.TBSCertList.RevokedCertificates
+	for i := range revoked {
+		if x509Cert.SerialNumber.Cmp(revoked[i].SerialNumber) == 0 {
+			// certificate is in revoked list, serials match
+			crlRes.RevocationTime = certList.TBSCertList.ThisUpdate
+			crlRes.Revoked = true
+
+			if err := in.DBHandle.UpdateCertMarkAsRevoked(in.Certificate.ID); err != nil {
+				resChan <- worker.Result{
+					Success:    false,
+					WorkerName: workerName,
+					Errors:     []string{fmt.Sprintf("error update Certificate %s revoked_via_crl, err=%v", in.Certificate.ID, err)},
+					Result:     nil,
+				}
+				return
+			}
+			break
+		}
+	}
+
+	// Marshal the response
+	bs, err := json.Marshal(&crlRes)
+	if err != nil {
+		result.Success = false
+		result.Errors = append(result.Errors, err.Error())
+	} else {
+		result.Success = true
+		result.Result = bs
+	}
+	resChan <- result
+}
+
+// Grab the first CRL response and return it in raw bytes
+func getCRLResponses(cert certificate.Certificate) ([][]byte, error) {
+	var wg sync.WaitGroup
+	var out [][]byte
+
+	crlPoints := cert.X509v3Extensions.CRLDistributionPoints
+	wg.Add(len(crlPoints))
+	for i := range crlPoints {
+		go func(point string, wg *sync.WaitGroup) {
+			defer wg.Done()
+
+			resp, err := httpClient.Get(point)
+			if err != nil {
+				return
+			}
+			defer resp.Body.Close()
+			bs, err := ioutil.ReadAll(resp.Body)
+			if err != nil {
+				return
+			}
+			out = append(out, bs)
+		}(crlPoints[i], &wg)
+	}
+	wg.Wait()
+
+	if len(out) > 0 {
+		return out, nil
+	}
+	return nil, fmt.Errorf("Unable to load CRL data for certificate %s", cert.Subject)
+}
+
+func verifyCRL(certList *pkix.CertificateList, cert certificate.Certificate, chain *certificate.Chain) error {
+	for i := range chain.Certs {
+		// each cert is a base64 DER encoded certificate
+		raw, err := base64.StdEncoding.DecodeString(chain.Certs[i])
+		if err != nil {
+			return fmt.Errorf("error decoding base64 DER of Certificate from %s chain, err=%v", chain.Domain, err)
+		}
+		cert, err := x509.ParseCertificate(raw)
+		if err != nil {
+			return fmt.Errorf("error reading certificate, err=%v", err)
+		}
+
+		// Check if cert signed the CRL response
+		if err := cert.CheckCRLSignature(certList); err == nil {
+			return nil // cert signed our CRL
+		}
+	}
+	return fmt.Errorf("Unable to verify CRL against %s chain", chain.Domain)
+}
+
+// AnalysisPrinter outputs results of comparing a certificate against the CRL(s) contained within
+func (eval) AnalysisPrinter(input []byte, printAll interface{}) (results []string, err error) {
+	var result Result
+	if err := json.Unmarshal(input, &result); err != nil {
+		return nil, fmt.Errorf("CRL Worker: failed to parse results: err=%v", err)
+	}
+
+	if result.Revoked {
+		results = append(results, fmt.Sprintf("* Certificate: Revoked (at %s)", result.RevocationTime.String()))
```

Comment:
nit: replace "Certificate" with "CRL", and remove the parenthesis around the timestamp